From eb66f5cbc42d3a1b151a45e594d40f1932e263bb Mon Sep 17 00:00:00 2001
From: Hardevsinh Palaniya <hardevsinh.palaniya@siliconsignals.io>
Date: Mon, 5 Jun 2023 14:14:08 +0530
Subject: [PATCH] Added changes related to display

- changes related to display init sequence
- changes related to display timing

Signed-off-by: Hardevsinh Palaniya <hardevsinh.palaniya@siliconsignals.io>

diff --git a/drivers/gpu/drm/panel/panel-sitronix-st7701.c b/drivers/gpu/drm/panel/panel-sitronix-st7701.c
index dd21170b83e6..f537db20afe6 100644
--- a/drivers/gpu/drm/panel/panel-sitronix-st7701.c
+++ b/drivers/gpu/drm/panel/panel-sitronix-st7701.c
@@ -29,45 +29,39 @@
 #include <drm/drm_print.h>
 #include <linux/delay.h>
 
-u8 dcs0[] = {0xFF ,0x77 ,0x01 ,0x00 ,0x00 ,0x13};
-u8 dcs1[] = {0xEF, 0x08};
-u8 dcs2[] = {0xFF, 0x77, 0x01, 0x00, 0x00, 0x10};
-u8 dcs3[] = {0xC0, 0xE9, 0x03};
-u8 dcs4[] = {0xC1, 0x10, 0x0C};
-u8 dcs5[] = {0xC2, 0x01, 0x0A};
-u8 dcs6[] = {0xC3, 0x02};
-u8 dcs7[] = {0xCC, 0x10};
-u8 dcs8[] = {0xCD, 0x08};
-u8 dcs9[] =  {0xB0, 0x0D, 0x14, 0x9C, 0x0B, 0x10, 0x06, 0x08, 0x09, 0x08, 0x22, 0x02, 0x4F, 0x0E, 0x66, 0x2D, 0x1F };
-u8 dcs10[] = {0xB1, 0x00, 0x17, 0x9E, 0x0F, 0x11, 0x06, 0x0C, 0x08, 0x08, 0x26, 0x04, 0x51, 0x10, 0x6A, 0x33, 0x1B };
-u8 dcs11[] = {0xFF, 0x77, 0x01, 0x00, 0x00, 0x11 };
-u8 dcs12[] = {0xB0, 0x30 };
-u8 dcs13[] = {0xB1, 0x67};
-u8 dcs14[] = {0xB2, 0x84};
-u8 dcs15[] = {0xB3, 0x80 };
-u8 dcs16[] = {0xB5, 0x4E };
-u8 dcs17[] = {0xB7, 0x85 };
-u8 dcs18[] = {0xB8, 0x20 };
-u8 dcs19[] = {0xC1, 0x78 };
-u8 dcs20[] = {0xC2, 0x78 };
-u8 dcs21[] = {0xD0, 0x88 };
-u8 dcs22[] = {0xE0, 0x00, 0x00, 0x02 };
-u8 dcs23[] = {0xE1, 0x06, 0xA0, 0x08, 0xA0, 0x05, 0xA0, 0x07, 0xA0, 0x00, 0x44, 0x44};
-u8 dcs24[] = {0xE2, 0x30, 0x30, 0x44, 0x44, 0x6E, 0xA0, 0x00, 0x00, 0x6E, 0xA0, 0x00, 0x00 };
-u8 dcs25[] = {0xE3, 0x00, 0x00, 0x33, 0x33 };
-u8 dcs26[] = {0xE4, 0x44, 0x44 };
-u8 dcs27[] = {0xE5, 0x0D, 0x69, 0x0A, 0xA0, 0x0F, 0x6B, 0x0A, 0xA0, 0x09, 0x65, 0x0A, 0xA0, 0x0B, 0x67, 0x0A, 0xA0 };
-u8 dcs28[] = {0xE6, 0x00, 0x00, 0x33, 0x33 };
-u8 dcs29[] = {0xE7, 0x44, 0x44 };
-u8 dcs30[] = {0xE8, 0x0C, 0x68, 0x0A, 0xA0, 0x0E, 0x6A, 0x0A, 0xA0, 0x08, 0x64, 0x0A, 0xA0, 0x0A, 0x66, 0x0A, 0xA0 };
-u8 dcs31[] = {0xE9, 0x36 , 0x00 };
-u8 dcs32[] = {0xEB, 0x00, 0x01, 0xE4, 0xE4, 0x44, 0x88, 0x40 };
-u8 dcs33[] = {0xED, 0xFF, 0x45, 0x67, 0xFA, 0x01, 0x2B, 0xCF, 0xFF, 0xFF, 0xFC, 0xB2, 0x10, 0xAF, 0x76, 0x54, 0xFF };
-u8 dcs34[] = {0xEF, 0x10, 0x0D, 0x04, 0x08, 0x3F, 0x1F };
-u8 dcs35[] = {0x3A, 0x66 };
-u8 dcs36[] = {0x11, 0x00};
-u8 dcs37[] = {0x29, 0x00};
-u8 dcs38[] = {0x35, 0x00};
+u8 dcs0[] = {0xFF, 0x77, 0x01, 0x00, 0x00, 0x13 };
+u8 dcs1[] = {0xEF, 0x08 };
+u8 dcs2[] = {0xFF, 0x77, 0x01, 0x00, 0x00, 0x10 };
+u8 dcs3[] = {0xC0, 0x3B, 0x00 };
+u8 dcs4[] = {0xC1, 0x12, 0x0A };
+u8 dcs5[] = {0xC2, 0x07, 0x03 };
+u8 dcs6[] = {0xCC, 0x10 };
+u8 dcs7[] = {0xB0, 0x0F, 0x11, 0x17, 0x15, 0x15, 0x09, 0x0C, 0x08, 0x08, 0x26, 0x04, 0x59, 0x16, 0x66, 0x2D, 0x1F };
+u8 dcs8[] = {0xB1, 0x0F, 0x11, 0x17, 0x15, 0x15, 0x09, 0x0C, 0x08, 0x08, 0x26, 0x04, 0x59, 0x16, 0x66, 0x2D, 0x1F };
+u8 dcs9[] =  {0xFF, 0x77, 0x01, 0x00, 0x00, 0x11 };
+u8 dcs10[] = {0xB0, 0x6D };
+u8 dcs11[] = {0xB1, 0x38 };
+u8 dcs12[] = {0xB2, 0x01 };
+u8 dcs13[] = {0xB3, 0x80 };
+u8 dcs14[] = {0xB5, 0x4E };
+u8 dcs15[] = {0xB7, 0x85 };
+u8 dcs16[] = {0xB8, 0x20 };
+u8 dcs17[] = {0xC1, 0x78 };
+u8 dcs18[] = {0xC2, 0x78 };
+u8 dcs19[] = {0xD0, 0x88 };
+u8 dcs20[] = {0xE0, 0x00, 0x00, 0x02 };
+u8 dcs21[] = {0xE1, 0x07, 0x00, 0x09, 0x00, 0x06, 0x00, 0x08, 0x00, 0x00, 0x33, 0x33 };
+u8 dcs22[] = {0xE2, 0x11, 0x11, 0x33, 0x33, 0xF6, 0x00, 0xF6, 0x00, 0xF6, 0x00, 0xF6, 0x00, 0x00 };
+u8 dcs23[] = {0xE3, 0x00, 0x00, 0x11, 0x11 };
+u8 dcs24[] = {0xE4, 0x44, 0x44 };
+u8 dcs25[] = {0xE5, 0x0F, 0xF3, 0x3D, 0xFF, 0x11, 0xF5, 0x3D, 0xFF, 0x0B, 0xEF,0x3D, 0xFF, 0x0D, 0xF1, 0x3D, 0xFF };
+u8 dcs26[] = {0xE6, 0x00, 0x00, 0x11, 0x11 };
+u8 dcs27[] = {0xE7, 0x44, 0x44 };
+u8 dcs28[] = {0xE8, 0x0E, 0xF2, 0x3D, 0xFF, 0x10, 0xF4, 0x3D, 0xFF, 0x0A, 0xEE, 0x3D, 0xFF, 0x0C, 0xF0, 0x3D, 0xFF };
+u8 dcs29[] = {0xE9, 0x36, 0x00 };
+u8 dcs30[] = {0xEB, 0x00, 0x01, 0xE4, 0xE4, 0x44, 0xAA, 0x10 };
+u8 dcs31[] = {0xED, 0xFF, 0x45, 0x67, 0xFA, 0x01, 0x2B, 0xCF, 0xFF, 0xFF, 0xFC, 0xB2, 0x10, 0xAF ,0x76, 0x54, 0xFF };
+u8 dcs32[] = {0x29 };
 
 unsigned char *ST7701S_CMD_Group[]={
 	dcs0,
@@ -103,12 +97,6 @@ unsigned char *ST7701S_CMD_Group[]={
 	dcs30,
 	dcs31,
 	dcs32,
-	dcs33,
-	dcs34,
-	dcs35,
-	dcs36,
-	dcs37,
-	dcs38,
 };
 unsigned long int ST7701S_CMD_Group_size[]={
 	sizeof(dcs0),
@@ -144,12 +132,6 @@ unsigned long int ST7701S_CMD_Group_size[]={
 	sizeof(dcs30),
 	sizeof(dcs31),
 	sizeof(dcs32),
-	sizeof(dcs33),
-	sizeof(dcs34),
-	sizeof(dcs35),
-	sizeof(dcs36),
-	sizeof(dcs37),
-	sizeof(dcs38),
 };
 
 static const u32 st7701s_bus_formats[] = {
@@ -187,7 +169,7 @@ static int st7701s_panel_push_cmd_list(struct mipi_dsi_device *dsi)
 	dsi->mode_flags |= MIPI_DSI_MODE_LPM;
 
 	//send init cmds
-	for(i=0; i<39; i++)
+	for(i=0; i<32; i++)
 	{
 		ret = mipi_dsi_generic_write(dsi, ST7701S_CMD_Group[i],
 						ST7701S_CMD_Group_size[i]);
@@ -197,7 +179,8 @@ static int st7701s_panel_push_cmd_list(struct mipi_dsi_device *dsi)
 			return ret;
 		}
 	}
-
+	mdelay(120);
+	mipi_dsi_generic_write(dsi, ST7701S_CMD_Group[32], ST7701S_CMD_Group_size[32] );
 	return ret;
 };
 
@@ -440,15 +423,15 @@ static const struct drm_panel_funcs st7701s_panel_funcs = {
  * to 132MHz (60Hz refresh rate)
  */
 static const struct display_timing st7701s_default_timing = {
-	.pixelclock = {8294400, 16588800, 16588800}, //Hz : htotal*vtotal*60 = 23 908 800 Hz
+	.pixelclock = {17500000, 19000000, 16632000}, //Hz : htotal*vtotal*60 = 23 908 800 Hz
 	.hactive = { 480, 480, 480 },
-	.hfront_porch = { 1, 1, 50 },
-	.hsync_len = { 1, 1, 10 },
-	.hback_porch = { 50, 50, 46 },
+	.hfront_porch = { 16, 48, 8 },
+	.hsync_len = { 40, 32, 32 },
+	.hback_porch = { 56, 80, 40 },
 	.vactive = { 480, 480, 480 },
-	.vfront_porch = { 10, 15, 15 },
-	.vsync_len = { 1, 1, 10 },
-	.vback_porch = { 5, 15, 15 },
+	.vfront_porch = { 3, 3, 1 },
+	.vsync_len = { 10, 10, 8 },
+	.vback_porch = { 7, 6, 6 },
 	.flags = DISPLAY_FLAGS_HSYNC_LOW |
 		 DISPLAY_FLAGS_VSYNC_LOW |
 		 DISPLAY_FLAGS_DE_LOW|
-- 
2.25.1

